l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -181,24 +205,49 @@ contract MailboxFacet is Base, IMailbox {
         uint16 _l2TxNumberInBatch,
         bytes calldata _message,
         bytes32[] calldata _merkleProof
-    ) external override nonReentrant {
-        require(!s.isEthWithdrawalFinalized[_l2BatchNumber][_l2MessageIndex], "jj");
-
-        L2Message memory l2ToL1Message = L2Message({
-            txNumberInBatch: _l2TxNumberInBatch,
-            sender: L2_ETH_TOKEN_SYSTEM_CONTRACT_ADDR,
-            data: _message
-        });
-
-        (address _l1WithdrawReceiver, uint256 _amount) = _parseL2WithdrawalMessage(_message);
-
-        bool proofValid = proveL2MessageInclusion(_l2BatchNumber, _l2MessageIndex, l2ToL1Message, _merkleProof);
-        require(proofValid, "pi"); // Failed to verify that withdrawal was actually initialized on L2
-
-        s.isEthWithdrawalFinalized[_l2BatchNumber][_l2MessageIndex] = true;
-        _withdrawFunds(_l1WithdrawReceiver, _amount);
+    ) external override {
+        require(
+            s.baseToken == ETH_TOKEN_ADDRESS,
+            " finalizeEthWithdrawal only available for Eth chains on mailbox"
+        );
+        IL1Bridge(s.baseTokenBridge).finalizeWithdrawal(
+            s.chainId,
+            _l2BatchNumber,
+            _l2MessageIndex,
+            _l2TxNumberInBatch,
+            _message,
+            _merkleProof
+        );
+    }
 
-        emit EthWithdrawalFinalized(_l1WithdrawReceiver, _amount);
+    // legacy for era
+    function requestL2Transaction(
+        address _contractL2,
+        uint256 _l2Value,
+        bytes calldata _calldata,
+        uint256 _l2GasLimit,
+        uint256 _l2GasPerPubdataByteLimit,
+        bytes[] calldata _factoryDeps,
+        address _refundRecipient
+    ) external payable returns (bytes32 canonicalTxHash) {
+        require(s.baseToken == ETH_TOKEN_ADDRESS, "legacy interface only available for eth base token");
+        canonicalTxHash = _requestL2TransactionSender(
+            msg.sender,
+            _contractL2,
+            msg.value,
+            _l2Value,
+            _calldata,
+            _l2GasLimit,
+            _l2GasPerPubdataByteLimit,
+            _factoryDeps,
+            _refundRecipient
+        );
+        IL1Bridge(s.baseTokenBridge).bridgehubDeposit{value: msg.value}(
+            s.chainId,
+            address(0),
```

Comment:
I don't understand yet whether you want to use weth bridge as eth bridge. But if so then it should be. 
```suggestion
            address(1),
```
<img width="745" alt="image" src="https://github.com/matter-labs/era-contracts/assets/41153528/af7144e2-1ab6-4e1e-bff7-6961dbfcabd5">

Reply:
Actually I'll just use the ETH_TOKEN_CONSTANT that I already made